### PR TITLE
sosreport.py: Taking the sosreport for different file systems

### PR DESCRIPTION
**Pull request by shirishaganta1**
This code creates different file systems and collects the sosreport

Signed-off-by: Shirisha Ganta <SHIRISHA.Ganta1@ibm.com>

### CODE REVIEW

**Comment by shirishaganta1** [comment]
JOB LOG    : /home/avocado-fvt-wrapper/results/job-2022-12-06T00.20-3a31da5/job.log
 (01/30) sosreport.py:Sosreport.test_short;run-filesystem-ext4-e656:  PASS (552.25 s)
 (02/30) sosreport.py:Sosreport.test_short;run-filesystem-xfs-4133:  PASS (552.54 s)
 (03/30) sosreport.py:Sosreport.test_short;run-filesystem-ext3-76a7:  PASS (555.44 s)
 (04/30) sosreport.py:Sosreport.test_user;run-filesystem-ext4-e656:  PASS (1648.11 s)
 (05/30) sosreport.py:Sosreport.test_user;run-filesystem-xfs-4133:  PASS (1649.00 s)
 (06/30) sosreport.py:Sosreport.test_user;run-filesystem-ext3-76a7:  PASS (1645.90 s)
 (07/30) sosreport.py:Sosreport.test_plugins;run-filesystem-ext4-e656:  PASS (553.19 s)
 (08/30) sosreport.py:Sosreport.test_plugins;run-filesystem-xfs-4133:  PASS (553.95 s)
 (09/30) sosreport.py:Sosreport.test_plugins;run-filesystem-ext3-76a7:  PASS (556.04 s)
 (10/30) sosreport.py:Sosreport.test_others;run-filesystem-ext4-e656:  PASS (995.77 s)
 (11/30) sosreport.py:Sosreport.test_others;run-filesystem-xfs-4133:  PASS (991.16 s)
 (12/30) sosreport.py:Sosreport.test_others;run-filesystem-ext3-76a7:  PASS (992.01 s)
 (13/30) sosreport.py:Sosreport.test_archive;run-filesystem-ext4-e656:  PASS (538.04 s)
 (14/30) sosreport.py:Sosreport.test_archive;run-filesystem-xfs-4133:  PASS (538.70 s)
 (15/30) sosreport.py:Sosreport.test_archive;run-filesystem-ext3-76a7:  PASS (540.72 s)
 (16/30) sosreport.py:Sosreport.test_PPC;run-filesystem-ext4-e656:  PASS (139.32 s)
 (17/30) sosreport.py:Sosreport.test_PPC;run-filesystem-xfs-4133:  PASS (138.14 s)
 (18/30) sosreport.py:Sosreport.test_PPC;run-filesystem-ext3-76a7:  PASS (138.22 s)
 (19/30) sosreport.py:Sosreport.test_smtchanges;run-filesystem-ext4-e656:  PASS (757.48 s)
 (20/30) sosreport.py:Sosreport.test_smtchanges;run-filesystem-xfs-4133:  PASS (752.32 s)
 (21/30) sosreport.py:Sosreport.test_smtchanges;run-filesystem-ext3-76a7:  PASS (752.06 s)
 (22/30) sosreport.py:Sosreport.test_dlpar_cpu_hotplug;run-filesystem-ext4-e656:  PASS (195.39 s)
 (23/30) sosreport.py:Sosreport.test_dlpar_cpu_hotplug;run-filesystem-xfs-4133:  PASS (191.27 s)
 (24/30) sosreport.py:Sosreport.test_dlpar_cpu_hotplug;run-filesystem-ext3-76a7:  PASS (191.58 s)
 (25/30) sosreport.py:Sosreport.test_dlpar_mem_hotplug;run-filesystem-ext4-e656:  PASS (186.21 s)
 (26/30) sosreport.py:Sosreport.test_dlpar_mem_hotplug;run-filesystem-xfs-4133:  PASS (187.00 s)
 (27/30) sosreport.py:Sosreport.test_dlpar_mem_hotplug;run-filesystem-ext3-76a7:  PASS (187.16 s)
 (28/30) sosreport.py:Sosreport.test_fs;run-filesystem-ext4-e656:  PASS (189.91 s)
 (29/30) sosreport.py:Sosreport.test_fs;run-filesystem-xfs-4133:  PASS (188.32 s)
 (30/30) sosreport.py:Sosreport.test_fs;run-filesystem-ext3-76a7:  PASS (189.74 s)
RESULTS    : PASS 30 | ERROR 0 | FAIL 0 | SKIP 0 | WARN 0 | INTERRUPT 0 | CANCEL 0
JOB HTML   : /home/avocado-fvt-wrapper/results/job-2022-12-06T00.20-3a31da5/results.html
JOB TIME   : 17273.36 s

**Review by abdhaleegit — CHANGES_REQUESTED** [review]
@shirishaganta1 @PraveenPenguin cant we use ltp_fs test instead adding code here ?

File: ras/sosreport.py [hunk 1]
```diff
@@ -363,3 +363,27 @@ def run_workload(self):
...
             genio.write_one_line(cpu_file, "1")
+
+    def test_fs(self):
+        is_fail = 0
+        loop_dev = "/dev/loop0"
+        fstype = self.params.get('fs', default='ext4')
+        mnt = self.params.get('dir', default='/mnt')
+        if 'blockfile' not in self.run_cmd_out("ls /tmp"):
+            blk_dev = process.run("dd if=/dev/zero of=/tmp/blockfile bs=1M count=5120")
+            create_dev = process.run("losetup %s /tmp/blockfile" % loop_dev)
```
Comment (sacsant) [inline ras/sosreport.py:374]: Is create_dev required? It's not used any where else.

File: ras/sosreport.py [hunk 2]
```diff
@@ -363,3 +363,27 @@ def run_workload(self):
...
+        if fstype == "ext4":
+            cmd = "mkfs.%s %s" % (fstype, loop_dev)
+        else:
+            cmd = "mkfs.%s -f %s" % (fstype, loop_dev)
+        process.run("mkfs.%s %s" % (fstype, loop_dev))
```
Comment (sacsant) [inline ras/sosreport.py:379]: cmd variable should be used here.

File: ras/sosreport.py.data/options.yaml [hunk 3]
```diff
@@ -5,3 +5,12 @@ list     : ['-o dlm -k dlm.lockdump', '-k filesys.dumpe2fs=on',
...
             'ps', 'sos_commands', 'sos_logs', 'sos_reports']
+
+dir: '/mnt'
+filesystem: !mux
+    ext4:
+        fs: 'ext4'
+    xfs:
+        fs: 'xfs'
+    ext3:
+        fs: 'ext3'
```
Comment (sacsant) [inline ras/sosreport.py.data/options.yaml:16]: Can you remove ext3 and instead add btrfs? Also make sure to add a check for mkfs.btrfs in the code. If this command is missing the test case should not error-out.

Comment (shirishaganta1) [reply ras/sosreport.py.data/options.yaml:16]: will remove ext3 and as discussed over slack will not add btrfs as it anyways not supported on Rhel. And also addressed the above comments

**Review by PraveenPenguin — APPROVED** [review]

**Pull request merged**